src/services/rfqm_service.ts

```diff
@@ -439,6 +450,118 @@ export class RfqmService {
         return Math.random() > 0.5 ? Promise.reject('failing') : Promise.resolve();
     }
 
+    /**
+     * Submit a transaction, re-submit if necessary, and poll for resolution
+     */
+    public async completeSubmissionLifecycleAsync(
+        orderHash: string,
+        workerAddress: string,
+        callData: string,
+    ): Promise<void> {
+        const submissions: { [transactionHash: string]: RfqmTransactionSubmissionEntity } = {};
```

Comment:
nit: can we name this `submissionMap` - I keep on thinking of `submissions` as a list